server/src/core/link-generator/link-generator-service.ts

```diff
@@ -1,8 +1,12 @@
 import { LinkGeneratorService, DonateLinkArgs } from "./types";
+import { BitlyLinkShortener } from './bitly-link-shortener';
 import { User } from '../user';
+import * as queryString from 'querystring';
+import { rethrowIfAppError } from '../error';
 
 interface LinkArgs { 
   baseUrl: string;
+  bitly: BitlyLinkShortener
```

Comment:
The LinkGenerator should not know what implementation of link shortener that's passed to it.
So change this to `shorter: LinkShorternerService`